src/utils/common.ts

```diff
@@ -1,2 +1,8 @@
 /** A type to turn any type optional properties to required */
 export type RequiredProperties<T> = { [K in keyof T]-?: T[K] };
+
+/** A generic type for native events */
+export type EventProps = {
+  onClick?: () => void;
```

Comment:
Can we pass the event or no ?

Reply:
on both events i mean

Reply:
Yes we can. But then it will be specific to button since the event is 
` React.MouseEvent<HTMLButtonElement, MouseEvent>`
instead of simply 
`React.MouseEvent`